linkerd/meshtls/rustls/src/creds/verify.rs

```diff
@@ -0,0 +1,56 @@
+use linkerd_identity as id;
+use std::convert::TryFrom;
+use std::time::SystemTime;
+use std::{io, sync::Arc};
+use tokio_rustls::rustls::{
+    self,
+    client::{self, ServerCertVerified, ServerCertVerifier},
+    server::ParsedCertificate,
+    Certificate, RootCertStore, ServerName,
+};
+use tracing::trace;
+
+pub(crate) struct AnySANVerifier(Arc<RootCertStore>);
+
+impl AnySANVerifier {
+    pub(crate) fn new(roots: impl Into<Arc<RootCertStore>>) -> Self {
+        Self(roots.into())
+    }
+}
+
+impl ServerCertVerifier for AnySANVerifier {
+    /// Will verify the certificate is valid in the following ways:
+    /// - Signed by a  trusted `RootCertStore` CA
+    /// - Not Expired
+    fn verify_server_cert(
+        &self,
+        end_entity: &Certificate,
+        intermediates: &[Certificate],
+        _: &ServerName,
+        _: &mut dyn Iterator<Item = &[u8]>,
+        ocsp_response: &[u8],
+        now: SystemTime,
+    ) -> Result<ServerCertVerified, rustls::Error> {
+        let cert = ParsedCertificate::try_from(end_entity)?;
+
+        client::verify_server_cert_signed_by_trust_anchor(&cert, &self.0, intermediates, now)?;
+
+        if !ocsp_response.is_empty() {
+            trace!("Unvalidated OCSP response: {:?}", ocsp_response.to_vec());
```

Comment:
we shouldn't need to turn the OCSP response into a `Vec` just to format it, AFAIK:
```suggestion
            trace!("Unvalidated OCSP response: {ocsp_response:?}");
```